packages/neuron-ui/src/components/NervosDAO/hooks.ts

```diff
@@ -280,42 +78,20 @@ export const useUpdateGlobalAPC = ({
     }
   }, [bestKnownBlockTimestamp, genesisBlockTimestamp, setGlobalAPC])
 
-export const useOnDepositDialogDismiss = ({
-  setShowDepositDialog,
-  setDepositValue,
-  setErrorMessage,
-}: {
-  setShowDepositDialog: React.Dispatch<React.SetStateAction<boolean>>
-  setDepositValue: React.Dispatch<React.SetStateAction<string>>
-  setErrorMessage: React.Dispatch<React.SetStateAction<string>>
-}) =>
-  useCallback(() => {
-    setShowDepositDialog(false)
-    setDepositValue(`${MIN_DEPOSIT_AMOUNT}`)
-    setErrorMessage('')
-  }, [setShowDepositDialog, setDepositValue, setErrorMessage])
-
-export const useOnDepositDialogSubmit = ({
-  setShowDepositDialog,
-  setDepositValue,
-  dispatch,
-  walletID,
-}: {
-  setShowDepositDialog: React.Dispatch<React.SetStateAction<boolean>>
-  setDepositValue: React.Dispatch<React.SetStateAction<string>>
-  dispatch: React.Dispatch<StateAction>
-  walletID: string
-}) =>
-  useCallback(() => {
+export const useDepositDialog = () => {
+  const [showDepositDialog, setShowDepositDialog] = useState(false)
+  const onOpenDepositDialog = useCallback(() => {
+    setShowDepositDialog(true)
+  }, [])
+  const onCloseDepositDialog = useCallback(() => {
     setShowDepositDialog(false)
-    dispatch({
-      type: AppActions.RequestPassword,
-      payload: {
-        walletID,
-        actionType: 'send',
-      },
-    })
-  }, [setShowDepositDialog, setDepositValue, dispatch, walletID])
+  }, [])
+  return {
```

Comment:
Lack of dependency: `setShowDepositDialog`

Reply:
`setShowDepositDialog` does not need to add to dependence.

Reply:
so it's unnecessary to use `useCallback`

Reply:
Using `useCallback` will avoid creating a new function when the component is re-render.